gpytorch/lazy/block_diag_lazy_tensor.py

```diff
@@ -118,6 +122,14 @@ def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True)
             logdet_res = logdet_res.view(*self.batch_shape, -1).sum(-1)
         return inv_quad_res, logdet_res
 
+    @cached(name="root_decomposition")
+    def root_decomposition(self):
+        if settings.fast_computations.covar_root_decomposition.on():
+            return super().root_decomposition()
```

Comment:
I imagine that what we probably should be doing for the standard case is:

`RootLazyTensor(self.__class__(self.base_lazy_tensor.root_decomposition().root, num_blocks=self.num_blocks))`

This should be faster and more accurate than what we're currently doing.

Reply:
sounds good, will update